js/views/onboardingModal.js

```diff
@@ -0,0 +1,438 @@
+'use strict';
+
+var Backbone = require('backbone'),
+    loadTemplate = require('../utils/loadTemplate'),
+    cropit = require('../utils/jquery.cropit'),
+    baseModal = require('./baseModal'),
+    messageModal = require('../utils/messageModal.js'),
+    countryListView = require('../views/countryListVw'),
+    currencyListView = require('../views/currencyListVw'),
+    languageListView = require('../views/languageListVw'),
+    guidStillCreatingModal = require('../views/guidStillCreatingModal');
+
+module.exports = baseModal.extend({
+  className: 'js-homeModal',
+
+  events: {
+    'click .js-homeModal': 'blockClicks',
+    'click .js-homeModal-countrySelect': 'countrySelect',
+    'click .js-homeModal-currencySelect': 'currencySelect',
+    'click .js-homeModal-languageSelect': 'languageSelect',
+    'click .js-homeModal-timeSelect': 'timeSelect',
+    'click .js-homeModal-newHandle': 'newHandle',
+    'click .js-homeModal-existingHandle': 'existingHandle',
+    'click .js-homeModal-cancelHandle': 'cancelHandle',
+    'click .js-accordionNext': 'accNext',
+    'click .js-accordionPrev': 'accPrev',
+    'keypress .js-accordionNext': 'accNextKeypress',
+    'keypress .js-accordionPrev': 'accPrevKeypress',
+    'click .js-homeModalDone': 'settingsDone',
+    'keypress .js-homeModalDone': 'settingsDoneKeypress',
+    'click .js-homeModal-themeSelected': 'setSelectedTheme',
+    'blur input, textarea': 'validateInput'
+  },
+
+  initialize: function(options) {
+    if (!(options.guidCreationPromise && options.guidCreationPromise.then)) {
+      throw new Error('Please provide a guidCreationPromise.');
+    }
+
+    this.options = options || {};
+    this.$document = $(document).on('focus', this.docFocusHandler);
+    this.$el.attr('tabIndex', 0);
+    this.$loadingModal = $('.js-loadingModal');
+  },
```

Comment:
You'll need to add a listener to listen for changes to the language in the user model, so that when someone selects a new language in onboarding, onboarding re-renders with the new language. 

Something like:

//when language is changed, re-render
      self.listenTo(self.model, 'change:language', function(){
        self.createTranslation(self.model.get("language"));
        self.render();
      });

  createTranslation: function(newLang){
    window.polyglot = new Polyglot({locale: newLang});
    window.polyglot.extend(__.where(this.languages.get('languages'), {langCode: newLang})[0]);
  },